src/framework/MOM_domains.F90

```diff
@@ -60,11 +63,12 @@ module MOM_domains
 !> MOM_domains_init initializes a MOM_domain_type variable, based on the information
 !! read in from a param_file_type, and optionally returns data describing various
 !! properties of the domain type.
-subroutine MOM_domains_init(MOM_dom, param_file, symmetric, static_memory, &
```

Comment:
Adding a mandatory `unit_scaling_type` argument to `MOM_domains_init()` is going give us problems with SIS2 being unable to compile with successive versions of MOM6, as noted by @kshedstrom.  Were MOM6 and SIS2 under the same version control, this would not be an issue, but unfortunately they are.  There is a simple solution, though.  If `US` were made into an optional argument to `MOM_domains_init()` and `gen_auto_mask_table()`, with `m_to_Z` defaulting to 1.0 when `US` is not present, everything here should work exactly as it does now, but SIS2 would be able to work across versions of the code and have the new `US` type argument added in an orderly way.
I would be happy to prepare a pull request to dev/ncar-candidate-240319 making these changes if it would help.

Reply:
Thanks, @Hallberg-NOAA! I'd appreciate it if you could submit a PR to dev/ncar-candidate-240319 with your proposed change.

Reply:
@alperaltuntas, I have submitted the PR to dev/ncar with my suggested changes at https://github.com/NCAR/MOM6/pull/272.  If these modifications are acceptable, I do not see any other issues within the larger PR from dev/ncar to main that I think will need to be considered.  I have run these changes through my version of the GFDL regression tests, and they are passing for me, although of course we will have to run the final revised version through our full suite of regression tests once it is available.

Reply:
Thanks! I just merged the changes.